### ISSUE TITLE
Picasso in main thread while offline

### ISSUE THREAD

**Issue by vernazza** [square/picasso#862]
I need to load a previously cached image (retrieved using `fetch()` in the main thread (Inside a `InfoWindowAdapter` in a MapView: I must fill it while `getInfoContents` is called, so I can't use callbacks or ausiliary threads).
I tried using
`Picasso.with(...).load(...).networkPolicy(NetworkPolicy.OFFLINE).get()`
but I get the exception
`java.lang.IllegalStateException: Method call should not happen from the main thread`

I understand why that happened, but I think that while using networkPolicy(NetworkPolicy.OFFLINE) Picasso should bybass that check

**Comment by JakeWharton** [comment]
Even when operating offline there is still a significant delay in reading the local image from disk. You will drop a few frames.

**Comment by necronet** [comment]
Any tip to improve the delay on loading a local image or is that a more okhttp issue?

**Comment by vernazza** [comment]
You are right JakeWarthon, but in some way, soon or later, I've to load it (and since I need a different image for every marker I can't preload in memory any image); if i'm lucky the image is cached in memory, if i'm not then I've to load it anyway (with or without picasso)
